client/handle_http.go

```diff
@@ -1120,25 +1184,124 @@ func runTransferWorker(ctx context.Context, workChan <-chan *clientTransferFile,
 	}
 }
 
+// If there are multiple potential attempts, try to see if we can quickly eliminate some of them
+//
+// Attempts a HEAD against all the endpoints simultaneously.  Put any that don't respond within
+// a second behind those that do respond.
+func sortAttempts(ctx context.Context, path string, attempts []transferAttemptDetails) (size int64, results []transferAttemptDetails) {
+	size = -1
+	if len(attempts) < 2 {
+		results = attempts
+		return
+	}
+	transport := config.GetTransport()
+	headChan := make(chan struct {
+		idx  int
+		size uint64
+		err  error
+	})
+	defer close(headChan)
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	for idx, transferEndpoint := range attempts {
+		tUrl := *transferEndpoint.Url
+		tUrl.Path = path
+
+		go func(idx int, tUrl string) {
+			headClient := &http.Client{Transport: transport}
+			headRequest, _ := http.NewRequestWithContext(ctx, "HEAD", tUrl, nil)
+			var headResponse *http.Response
+			headResponse, err := headClient.Do(headRequest)
+			if err != nil {
+				headChan <- struct {
+					idx  int
+					size uint64
+					err  error
+				}{idx, 0, err}
+				return
+			}
+			headResponse.Body.Close()
+			contentLengthStr := headResponse.Header.Get("Content-Length")
+			size := int64(0)
+			if contentLengthStr != "" {
+				size, err = strconv.ParseInt(contentLengthStr, 10, 64)
+				if err != nil {
+					log.Errorln("problem converting content-length to an int:", err)
+				}
+			}
+			headChan <- struct {
+				idx  int
+				size uint64
+				err  error
+			}{idx, uint64(size), nil}
```

Comment:
Copy comment from hidden commit history:

> Do we want to return error or nil here if there is an error converting Content-Length header?

Reply:
Good idea.  An error here will sort the cache toward the back (but not be fatal to the transfer).